DP-18/Add basic troubleshooting section to the docs (#5901)
OpenNMS/opennms: docs/modules/troubleshooting/pages/log-information.adoc

= Trace {page-component-title} Logs

This section describes common log files to look at when troubleshooting.

[[manager.log]]
== manager.log

`$OPENNMS_HOME/logs/manager.log` contains information relating to manager server events and the start and stop processes for {page-component-title} services.

Below is an example of a log that describes a problem connecting to the PostgreSQL database server:

[source, shell]
----
2023-03-20 14:08:11,256 INFO  [Main] o.o.f.s.j.JCEKSSecureCredentialsVault: No existing keystore found at: /opt/opennms/etc/scv.jce. Using empty keystore.
2023-03-20 14:08:11,257 ERROR [Main] o.o.n.v.DatabaseChecker: Unable to connect to data source 'opennms' at URL 'jdbc:postgresql://localhost:5432/opennms' with username 'opennms', check opennms-datasources.xml and your database permissions.
2023-03-20 14:08:11,258 ERROR [Main] o.o.n.v.Controller: error invoking "check" operation: Data source 'opennms' failed.
org.opennms.netmgt.vmmgr.InvalidDataSourceException: Data source 'opennms' failed.
----

[[karafStartupMonitor.log]]
== karafStartupMonitor.log

> `$OPENNMS_HOME/logs/karafStartupMonitor.log` contains information relating to server events and the start and stop processes for {page-component-title} karaf related services.

Reviewer: ```suggestion
`$OPENNMS_HOME/logs/karafStartupMonitor.log` contains information relating to startup of the [Karaf container](https://karaf.apache.org/) embedded within {page-component-title}, and of functionality that is implemented as Karaf features.
```

= {page-component-title} Log Files

This section describes common log files to look at when troubleshooting.

[[manager.log]]
== manager.log

> `$\{OPENNMS_HOME}/logs/manager.log` contains information relating to manager server events and the start and stop processes for {page-component-title} services.
> 
> For example, this log describes a problem connecting to the PostgreSQL database server:

Reviewer 2: ```suggestion
`$\{OPENNMS_HOME}/logs/manager.log` contains information relating to manager server events and the start and stop processes for {page-component-title} services.
For example, this log describes a problem connecting to the PostgreSQL database server:
```

[source, shell]
----
2023-03-20 14:08:11,256 INFO  [Main] o.o.f.s.j.JCEKSSecureCredentialsVault: No existing keystore found at: /opt/opennms/etc/scv.jce. Using empty keystore.
2023-03-20 14:08:11,257 ERROR [Main] o.o.n.v.DatabaseChecker: Unable to connect to data source 'opennms' at URL 'jdbc:postgresql://localhost:5432/opennms' with username 'opennms', check opennms-datasources.xml and your database permissions.
2023-03-20 14:08:11,258 ERROR [Main] o.o.n.v.Controller: error invoking "check" operation: Data source 'opennms' failed.
org.opennms.netmgt.vmmgr.InvalidDataSourceException: Data source 'opennms' failed.
----

[[karafStartupMonitor.log]]
== karafStartupMonitor.log

> `$\{OPENNMS_HOME}/logs/karafStartupMonitor.log` contains information relating to startup of the {page-component-title} Karaf container and its feature functionality.
> 
> For example, this log describes a problem with the Karaf container's startup process, and steps you can take to troubleshoot the error:

Reviewer 2: ```suggestion
`$\{OPENNMS_HOME}/logs/karafStartupMonitor.log` contains information relating to startup of the {page-component-title} Karaf container and its feature functionality.
For example, this log describes a problem with the Karaf container's startup process, and steps you can take to troubleshoot the error:
```
